cachi2/core/package_managers/yarn_classic/resolver.py

```diff
@@ -203,6 +207,25 @@ def _get_main_package(source_dir: RootedPath) -> WorkspacePackage:
     )
 
 
+def _get_workspace_packages(
+    source_dir: RootedPath, workspaces: list[Workspace]
+) -> list[WorkspacePackage]:
+    """Return a WorkspacePackage for each Workspace."""
+    return [
+        WorkspacePackage(
+            name=ws.package_contents["name"],
+            version=ws.package_contents.get("version"),
+            relpath=ws.path.relative_to(source_dir.path),
+        )
+        for ws in workspaces
+    ]
+
+
 def resolve_packages(source_dir: RootedPath) -> list[YarnClassicPackage]:
     """Return a list of Packages corresponding to all project dependencies."""
-    return [_get_main_package(source_dir)] + _get_packages_from_lockfile(source_dir)
+    workspaces = extract_workspace_metadata(source_dir)
+    return (
+        [_get_main_package(source_dir)]
+        + _get_workspace_packages(source_dir, workspaces)
+        + _get_packages_from_lockfile(source_dir)
+    )
```

Comment:
minor nitpick (originating in an earlier commit) : overoptimization suggestion - do we need to create yet another intermediary list in the memory for something that is going to be iterated once only for SBOM components creation? IOW isn't `itertools.chain` a slightly better choice here instead of list concatenation?